src/piece.c

```diff
@@ -262,23 +262,36 @@ pawn_valid(position pos, cell game[SIZE_STD][SIZE_STD], uint8_t * game_flags)
 	int		side_sign = 1;
 	if (game[pos.rank][pos.file].side == BLACK)
 		side_sign = -1;
-	if (game[pos.rank + side_sign][pos.file].piece == NULL)
+	if (game[pos.rank + side_sign][pos.file].piece == NULL 
+		&& pos.rank + side_sign >= 0 
+		&& pos.file >= 0 
```

Comment:
No need to check the file as pawn cannot move horizontally (only when capturing, see next comment)

Reply:
understood thanks